src/main/java/ru/yandex/clickhouse/response/ByteFragmentUtils.java

```diff
@@ -4,11 +4,23 @@
 import com.google.common.primitives.Primitives;
 
 import java.math.BigInteger;
+import java.sql.Date;
+import java.sql.Timestamp;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.TimeZone;
 
 public final class ByteFragmentUtils {
 
     public static final char ARRAY_ELEMENTS_SEPARATOR = ',';
     public static final char STRING_QUOTATION = '\'';
+    public static final DateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd");
```

Comment:
Anyway, you can't set timezone to static fields (because there may be several connections with different timezones), so you probably need to pass the formats from outside. As I mentioned in another comment, we already have SimpleDateFormats as object fields of ClickHouseResultSet with timezones already set, so it must be convenient to pass them to ByteFragmentUtils methods. This solution must be safe, at least it conforms with absence of thread safety in single ResultSet but doesn't allow different ResultSets to interfere.